File: examples/Scala/MsgQueue.scala
s/**
 * Simple message queuing broker
 * Same as request-reply broker but using QUEUE device.
 *
 * @author Giovanni Ruggiero
 * @email giovanni.ruggiero@gmail.com
 */

import org.zeromq.ZMQ
import org.zeromq.ZMQ.Context
import org.zeromq.ZMQ.Socket
import org.zeromq.ZMQQueue

object MsgQueue {
  def main(args : Array[String]) {

		//  Prepare our context and sockets
		val context = ZMQ.context(1)

		//  Socket facing clients
		val frontend = context.socket(ZMQ.ROUTER)
		frontend.bind("tcp://*:5559")

		//  Socket facing services
		val backend = context.socket(ZMQ.DEALER)
		backend.bind("tcp://*:5560")

		//  Start built-in device
		val queue = new ZMQQueue(context, frontend, backend)
		// have fun!

		//  We never get here but clean up anyhow
		frontend close

		backend close

		context term
	}
}

File: examples/Python/msgqueue.py
"""

   Simple message queuing broker
   Same as request-reply broker but using QUEUE device
 
   Author: Guillaume Aubert (gaubert) <guillaume(dot)aubert(at)gmail(dot)com>
  
"""


import zmq

def main():
    """ main method """
    
    context = zmq.Context(1)
    
    # Socket facing clients
    frontend = context.socket(zmq.ROUTER)
    frontend.bind("tcp://*:5559")
    
    # Socket facing services
    backend  = context.socket(zmq.DEALER)
    backend.bind("tcp://*:5560")
    
    zmq.device(zmq.QUEUE, frontend, backend)
    
    # We never get here...
    frontend.close()
    backend.close()
    context.term()
    


if __name__ == "__main__":
    main()
